annotations/src/lib.rs

```diff
@@ -73,11 +105,19 @@ pub enum TagPropagation {
     UninterpretedCall,
 }
 
+impl TagPropagation {
+    const fn into_set(self) -> TagPropagationSet {
+        1 << (self as u8)
+    }
+}
+
 /// Provide a way to create tag propagation sets. It is equivalent to bitwise-or of all its arguments.
 #[macro_export]
 macro_rules! tag_propagation_set {
-    ($($x:expr),*) => {
-        0 $(| (1 << ($x as u8)))*
+    () => { 0 };
+    ($x:expr) => { $x.into_set() };
+    ($x:expr,$($xs:expr),*) => {
+        add_propagation(tag_propagation_set!($($xs),*), $x)
```

Comment:
This is problematic since it requires the user of tag_propagation_set to also include add_propagation to its use statement. For example, see examples/tag_analysis/timing_channels/src/lib.rs:22:46.

It is not obvious to me that these changes actually add functionality.

Reply:
You are right. I will fix this. 
These changes were added for consistency, so that the logic for adding elements to tag propagation sets is not repeated in case it is ever changed.
I can revert this code or I can make the called function statically qualified using `$crate`. Which one would you prefer?

Reply:
Reverting the code seems simpler for now. I'm always in favor of simpler.